packages/types/src/types.ts

```diff
@@ -86,6 +86,11 @@ export interface Codec {
    */
   isEmpty: boolean;
 
+  /**
+   * @description The registry associated with this object
+   */
+  readonly registry: Registry;
```

Comment:
I don't think users should do ever `api.type('u32',2 ).registry`, would argue to remove this and put protected in class

Reply:
Ok, won't remove, but can add protected since it is only internally to the object and it's derivates.

Reply:
... adding `protected` is not actually allowed in the interface definition. It is exceptionally useful having it as "Codec", since we need to ensure that all types do implement it. so, rock, hard place.